gpflow/likelihoods/base.py

```diff
@@ -289,62 +292,75 @@ def _variational_expectations(self, Fmu, Fvar, Y):
 
 
 class QuadratureLikelihood(Likelihood):
-    def __init__(self, num_gauss_hermite_points: int = 20, **kwargs):
-        super().__init__(**kwargs)
-        self.num_gauss_hermite_points = num_gauss_hermite_points
-        self._quadrature = None
+    def __init__(
+        self,
+        latent_dim: int,
+        observation_dim: int,
+        quadrature: Optional[GaussianQuadrature] = None,
+        *,
+        num_gauss_hermite_points: int = DEFAULT_NUM_GAUSS_HERMITE_POINTS,
+    ):
+        super().__init__(latent_dim=latent_dim, observation_dim=observation_dim)
+        if quadrature is None:
+            with tf.init_scope():
+                quadrature = NDiagGHQuadrature(self._quadrature_dim, num_gauss_hermite_points)
+        self.quadrature = quadrature
 
     @property
-    def quadrature(self):
-        if self._quadrature is None:
-            if self.latent_dim is None:
-                raise Exception(
-                    "latent_dim not specified. "
-                    "Either set likelihood.latent_dim directly or "
-                    "call a method which passes data to have it inferred."
-                )
-            with tf.init_scope():
-                self._quadrature = NDiagGHQuadrature(self.latent_dim, self.num_gauss_hermite_points)
-        return self._quadrature
+    def _quadrature_dim(self) -> int:
+        return self.latent_dim
 
-    def _predict_mean_and_var(self, Fmu, Fvar):
+    def _quadrature_log_prob(self, F, Y):
+        return tf.expand_dims(self.log_prob(F, Y), axis=-1)
+
+    def _quadrature_reduction(self, v):
```

Comment:
As `ScalarLikelihood` overwrites this method, does it make sense to have it private? As it seems to be important for inheriting classes, I think a docstring would be nice.

Reply:
I don't think *other* classes would have to overwrite it, it's just because ScalarLikelihood behaves quite differently from the others ("broadcasting" over output dimensions and coupling latent_dim=observation_dim, without explicitly specifying either).

Reply:
Can you write this down in the code somewhere - maybe where you overwrite them.

Reply:
@vdutor @gustavocmv I've added docstrings, please check if a) they make sense and b) they satisfy your concerns!

Reply:
Checked them, lgtm.